### PR TITLE
【new: feature】Support the use of request Mapping to follow the solution

### PR DESCRIPTION
**Pull request by li-keguo**
about issue #2607

work list:

- [x] [shenyu-client-springmvc]The client supports the path in the mvc annotation and is backward compatible.
- [x] [shenyu-client-springmvc] clean a little warning.
- [x] [shenyu-example-http] add a little examples.

### CODE REVIEW

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java [hunk 1]
```diff
@@ -71,66 +89,186 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
-        String contextPath = this.contextPath;
-        String appName = this.appName;
-        String path;
-        if (StringUtils.isEmpty(contextPath)) {
-            path = prePath + shenyuSpringMvcClient.path();
-        } else {
-            path = contextPath + prePath + shenyuSpringMvcClient.path();
+    
+    /**
+     * Build api super path.<br>
```
Comment (yu199195) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java:210]: pls remove this doc， private method donot have doc。

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java [hunk 2]
```diff
@@ -71,66 +89,186 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
+        RequestMapping requestMapping = AnnotationUtils.findAnnotation(method, RequestMapping.class);
+        // Only the first path is supported temporarily
+        if (Objects.nonNull(requestMapping) && ArrayUtils.isNotEmpty(requestMapping.path()) && StringUtils.isNotBlank(requestMapping.path()[0])) {
+            return requestMapping.path()[0];
+        }
+        return "";
+    }
+    
+    /**
+     * Splicing path: splicing multiple paths into one path.<br>
```
Comment (yu199195) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java:232]: remove this

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java [hunk 3]
```diff
@@ -71,66 +89,186 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
+            }
+            result.append(p.startsWith(PATH_SEPARATOR) ? p.replaceFirst(PATH_SEPARATOR, "") : p);
+        }
+        return result.toString();
+    }
+    
+    /**
+     * build api metadata.
+     *
+     * @param shenyuSpringMvcClient api metadata resource
```
Comment (yu199195) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientBeanPostProcessor.java:257]: remove this

File: shenyu-examples/shenyu-examples-http/src/main/java/org/apache/shenyu/examples/http/controller/ShenyuClientPathController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * hello. <br>
+     * no support gateway.
+     *
+     * @param name say hello user name
+     * @return result
+     */
+    @RequestMapping("shenyu/client/hi")
+    public ResponseEntity<String> hello(final String name) {
+        return ResponseEntity.ok("hi! " + name + "! " + HELLO_SUFFIX);
```
Comment (yu199195) [inline shenyu-examples/shenyu-examples-http/src/main/java/org/apache/shenyu/examples/http/controller/ShenyuClientPathController.java:54]: donot used ResponseEntity.ok, pls see cve-79

Comment (li-keguo) [reply shenyu-examples/shenyu-examples-http/src/main/java/org/apache/shenyu/examples/http/controller/ShenyuClientPathController.java:54]: ok

**Review by yu199195 — APPROVED** [review]

**Pull request merged**